dv/verilator/memutil/cpp/verilator_memutil.cc

```diff
@@ -271,48 +275,59 @@ bool VerilatorMemUtil::ElfFileToBinary(const std::string &filepath,
     goto return_elf_end;
   }
 
-  GElf_Phdr phdr;
-  Elf_Data *elf_data;
-  elf_data = NULL;
-  for (size_t i = 0; i < phnum; i++) {
+  /*
+   * To mimic what objcopy does (that is, the binary target of BFD), we need to
+   * iterate over all loadable program headers, find the lowest address, and
+   * then copy in our loadable sections based on their offset with respect to
+   * the found base address.
+   */
+  for (i = 0; i < phnum; i++) {
     if (gelf_getphdr(elf_desc, i, &phdr) == NULL) {
       std::cerr << elf_errmsg(-1) << " segment number: " << i
                 << " in: " << filepath << std::endl;
       retval = false;
       goto return_elf_end;
     }
+
     if (phdr.p_type != PT_LOAD) {
-      std::cout << "Program header number " << i << "is not of type PT_LOAD."
-                << "Continue." << std::endl;
+      std::cout << "Program header number " << i << " is not of type PT_LOAD; "
+                << "ignoring." << std::endl;
       continue;
     }
+
+    if (phdr.p_filesz == 0)
```

Comment:
We're trying to follow the [OpenTitan C/C++ style guide](https://docs.opentitan.org/doc/rm/c_cpp_coding_style/) here which requires curly braces even for single-line statements. Can you add them here and in the statements below?

Reply:
This isn't mentioned in the README at the moment. I'll add it and get clang-format added to CI.

Reply:
Fixed -- and worth adding to [CONTRIBUTING.md](https://github.com/lowRISC/ibex/blob/master/CONTRIBUTING.md) as well?